app/models/fe/question_sheet.rb

```diff
@@ -34,24 +34,20 @@ def self.new_with_page
 
     # count all questions including ones inside a grid
     def questions_count
-      parent_ids = elements.find_all{ |e| e.is_a?(Fe::QuestionGrid) }.collect(&:id)
-      count = questions.count
-      # grab all sub-elements in one query to speed things up a bit
-      while (sub_elements = Fe::Element.where(question_grid_id: parent_ids)).any?
-        count += sub_elements.to_a.count{ |e| e.is_a?(Fe::Question) }
-        parent_ids = sub_elements.find_all{ |e| e.is_a?(Fe::QuestionGrid) }.collect(&:id)
-      end
-      return count
-    end
-
-    def questions
-      pages.collect(&:questions).flatten
+      all_elements.questions.count
     end
 
+    # returns all elements including ones inside a grid
     def elements
       pages.collect(&:elements).flatten
     end
 
+    def all_elements
+      element_ids = pages.pluck(:all_element_ids).join(',').split(',')
+      element_ids << 0 # in case there are no elements, the query breaks at IN ()
+      Element.where(id: element_ids.split(','))
```

Comment:
Thanks, will fix this.  No error because it just gives a warning:

2.1.3 :005 > Fe::Element.where(id: [[46]])
DEPRECATION WARNING: Passing a nested array to Active Record finder methods is deprecated and will be removed. Flatten your array before using it for 'IN' conditions. (called from irb_binding at (irb):5)